app/controllers/schedules_controller.rb

```diff
@@ -128,5 +142,14 @@ def schedule_params
                     tasks_attributes: [:id, :type, :sprout, :equipment, :duration, :trigger,
                                        :target_temperature, :ramp_estimate, :parent_id, :_destroy])
     end
+
+    def check_root_task(task_attributes)
```

Comment:
This should really be a validation in the model rather than a controller method.

Here's what I think you need to do to make that happen:
* Add another validation to Schedule that forces there to only be one Task on #new even outside of the controller. It should be something like:
```ruby
validates_numericality_of :tasks, { equal_to: 1 }, on: :new
```
* Add an if-statement around lines 19-27 in app/views/schedules/_fields.html.erb so that the button to add another task isn't rendered. When combined with your other guard around the button to remove tasks, I think that'll make it so you have one and only one task on the #new view, which is basically what you're trying to do on line 150.
* Remove the `on: :update` qualifier from the :root_task in the Schedule model. That'll replace the check you're trying to do on line 148 by requiring a root_task all the time.

Once you've done that, you can drop this method and drop the change on lines 63-70.